sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RuleDescriptionSection.java

```diff
@@ -46,8 +48,22 @@ private RuleDescriptionSectionKeys() {
 
   String getHtmlContent();
 
+  default Optional<Context> getContext() { return Optional.empty(); }
+
+  /**
+   * @deprecated use {@link #ruleDescriptionSectionBuilder()} instead.
+   */
+  @Deprecated(since = "9.7", forRemoval = true)
   static RuleDescriptionSectionBuilder builder() {
```

Comment:
Why not keep `builder()` and add `contextAwareBuilder()`?

Reply:
Ok, I went for it!